chess/board.py

```diff
@@ -0,0 +1,178 @@
+        # Rooks
+        self.matrix[j][0] = Rook(color)
+        self.matrix[j][7] = Rook(color)
+        # Knights
+        self.matrix[j][1] = Knight(color)
+        self.matrix[j][6] = Knight(color)
+        # Bishops
+        self.matrix[j][2] = Bishop(color)
+        self.matrix[j][5] = Bishop(color)
+        # Royalty
+        self.matrix[j][3] = Queen(color)
+        self.matrix[j][4] = King(color)
+
+    def __getitem__(self, pos: str) -> Optional[Piece]:
+        """
+        Args:
+            pos: Position on the chess board. The format of the string is
+                f"{letter}{number}", e.g. "A7".
+
+        Returns:
+            The Piece at the position on the Board
+        """
+        check_position_format(pos)
+        row = int(pos[1]) - 1
+        column = Board.columns[pos[0]]
+        return self.matrix[row][column]
+
+    def __setitem__(self, pos: str, piece: Piece) -> None:
+        """
+        Args:
+            pos: Position on the chess board. The format of the string is
+                f"{letter}{number}", e.g. "A7".
+            piece: the Piece to put on the board
+        """
+        check_position_format(pos)
+        row = int(pos[1]) - 1
+        column = Board.columns[pos[0]]
+        self.matrix[row][column] = piece
+
+    def inside_board(self, i: int, j: int) -> bool:
+        """Returns whether the indices are valid positions on the board"""
+        assert isinstance(i, int) and isinstance(j, int)
+        return 0 <= i < 8 and 0 <= j < 8
+
+    def idxs_to_position(self, i: int, j: int) -> str:
+        """Given a set of matrix indices, turn them into a Chess position.
+
+        Args:
+            i,j : Indices representing the chess Board
+
+        Returns:
+            corresponding Chess Postion, e.g. "A7"
+
+        """
+        row = str(i + 1)
+        col = Board.inv_cols[j]
+        pos = f"{col}{row}"
+        check_position_format(pos)
+        return pos
+
+    def position_to_idxs(self, pos: str) -> Tuple[int, int]:
+        """Converts Position to matrix indices.
+
+        Args:
+            pos: Position on the chess board. The format of the string is
+                f"{letter}{number}", e.g. "A7".
+
+        Returns:
+            Tuple of matrix indices
+        """
+        check_position_format(pos)
+        i, j = int(pos[1]) - 1, Board.columns[pos[0]]
+        return (i, j)
+
+    def valid_moves(self, pos: str) -> List[str]:
+        """Returns all of the valid moves of a Piece at a given Position.
+
+        Args:
+            pos: Position on the chess board. The format of the string is
+                f"{letter}{number}", e.g. "A7".
+
+        Returns:
+            List of valid positions the Piece at the given Position can move
+        """
+        curr_piece = self[pos]
+        moves = []
+        if isinstance(curr_piece, Pawn):
+            moves = self._pawn_valid_moves(pos)
+        return sorted(moves)
+
+    def _pawn_valid_moves(self, pos: str) -> List[str]:
```

Comment:
functions start with an underscore indicate that they should not be called by the class itself, but only within a function (tldr; indicates this is a  helper method)